internal/integration/api/selinux.go

```diff
@@ -64,6 +67,92 @@ func (suite *SELinuxSuite) getLabel(nodeCtx context.Context, pid int32) string {
 	return string(bytes.TrimSpace(value))
 }
 
+// TestRuntimeFileLabels reads labels of runtime-created files from xattrs
+// to ensure SELinux labels for files are set when they are created.
+func (suite *SELinuxSuite) TestRuntimeFileLabels() {
+	workers := suite.DiscoverNodeInternalIPsByType(suite.ctx, machine.TypeWorker)
+	controlplanes := suite.DiscoverNodeInternalIPsByType(suite.ctx, machine.TypeControlPlane)
+
+	expectedLabelsWorker := map[string]string{
+		constants.APIRuntimeSocketPath:  constants.APIRuntimeSocketLabel,
+		constants.APISocketPath:         constants.APISocketLabel,
+		constants.DBusClientSocketPath:  constants.DBusClientSocketLabel,
+		constants.UdevRulesPath:         constants.UdevRulesLabel,
+		constants.DBusServiceSocketPath: constants.DBusServiceSocketLabel,
+		constants.MachineSocketPath:     constants.MachineSocketLabel,
+	}
+
+	expectedLabelsControlPlane := map[string]string{
+		constants.APIRuntimeSocketPath:  constants.APIRuntimeSocketLabel,
+		constants.APISocketPath:         constants.APISocketLabel,
+		constants.DBusClientSocketPath:  constants.DBusClientSocketLabel,
+		constants.UdevRulesPath:         constants.UdevRulesLabel,
+		constants.DBusServiceSocketPath: constants.DBusServiceSocketLabel,
+		constants.MachineSocketPath:     constants.MachineSocketLabel,
+		// Only running on controlplane
+		constants.EtcdPKIPath:                           constants.EtcdPKISELinuxLabel,
+		constants.KubernetesAPIServerConfigDir:          constants.KubernetesAPIServerConfigDirSELinuxLabel,
+		constants.KubernetesAPIServerSecretsDir:         constants.KubernetesAPIServerSecretsDirSELinuxLabel,
+		constants.KubernetesControllerManagerSecretsDir: constants.KubernetesControllerManagerSecretsDirSELinuxLabel,
+		constants.KubernetesSchedulerConfigDir:          constants.KubernetesSchedulerConfigDirSELinuxLabel,
+		constants.KubernetesSchedulerSecretsDir:         constants.KubernetesSchedulerSecretsDirSELinuxLabel,
+		constants.TrustdRuntimeSocketPath:               constants.TrustdRuntimeSocketLabel,
+	}
+
+	suite.checkFileLabels(workers, expectedLabelsWorker)
+	suite.checkFileLabels(controlplanes, expectedLabelsControlPlane)
+}
+
+func (suite *SELinuxSuite) checkFileLabels(nodes []string, expectedLabels map[string]string) {
+	for _, node := range nodes {
+		nodeCtx := client.WithNode(suite.ctx, node)
+		cmdline := suite.ReadCmdline(nodeCtx)
+
+		seLinuxEnabled := pointer.SafeDeref(procfs.NewCmdline(cmdline).Get(constants.KernelParamSELinux).First()) != ""
```

Comment:
this can be moved in setuptest so we can skip earlier

Reply:
I do it here to check per-node, as we might have different cmdline on each node